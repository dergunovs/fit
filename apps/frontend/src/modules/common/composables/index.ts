import { ref, computed, onMounted } from 'vue';
import { useRouter, useRoute } from 'vue-router';

import LayoutAdmin from '@/common/components/LayoutAdmin.vue';
import LayoutDefault from '@/common/components/LayoutDefault.vue';

export function useLayout() {
  const router = useRouter();
  const route = useRoute();

  const isLoaded = ref(false);

  const layoutComponent = computed(() => (route.meta.layout === 'admin' ? LayoutAdmin : LayoutDefault));

  onMounted(async () => {
    await router.isReady();
    isLoaded.value = true;
  });

  return {
    isLoaded,
    layoutComponent,
  };
}

export function useRouteId(title: string, isQuery?: boolean) {
  const route = useRoute();

  const id = computed(() => {
    const value = isQuery ? route.query[title]?.toString() : route.params[title].toString();

    return value || '';
  });

  return { id };
}
