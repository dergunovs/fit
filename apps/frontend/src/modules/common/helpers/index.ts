import { toast } from 'mhz-ui';

export async function copyToClipboard(text: string, toastText: string) {
  await navigator.clipboard.writeText(text);

  toast.success(toastText);
}
