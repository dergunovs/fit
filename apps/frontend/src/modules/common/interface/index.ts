import { TPostAuthLoginDTO } from 'fitness-tracker-contracts';
import { FunctionalComponent } from 'vue';

export interface INavItem {
  url: string;
  title: string;
  icon: FunctionalComponent;
}

export interface IOnSuccess {
  create?: (id?: string) => Promise<void>;
  update?: () => Promise<void>;
  delete?: () => Promise<void>;
  login?: (user: TPostAuthLoginDTO) => Promise<void>;
  setup?: () => Promise<void>;
  register?: () => Promise<void>;
}

export interface IPromptEvent extends Event {
  prompt: () => Promise<void>;
}

export interface IGetLatestVersionDTO {
  'dist-tags': { latest: string };
}
