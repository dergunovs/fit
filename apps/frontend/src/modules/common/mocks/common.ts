import { vi } from 'vitest';

export const spyCopyToClipboard = vi.spyOn(navigator.clipboard, 'writeText').mockResolvedValue();
