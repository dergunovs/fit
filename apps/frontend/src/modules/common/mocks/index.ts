export * from './helpers';
export * from './ui';
export * from './router';
export * from './layout';
export * from './pwa';
export * from './service';
