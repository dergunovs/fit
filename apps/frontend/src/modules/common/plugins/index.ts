import { createI18n, I18nOptions } from 'vue-i18n';

import ru from '@/common/locales/ru.json';
import en from '@/common/locales/en.json';

const messages = { en, ru } as const;

const options: I18nOptions = { legacy: false, locale: 'ru', messages };

// eslint-disable-next-line @typescript-eslint/no-type-alias
export type TMessageSchema = typeof en;

export const i18n = createI18n<false, typeof options>(options);
