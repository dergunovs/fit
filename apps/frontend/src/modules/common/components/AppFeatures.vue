<template>
  <div :class="$style.features">
    <div v-for="feature in APP_FEATURES" :key="feature.title" :class="$style.feature" data-test="app-feature">
      <component :is="feature.icon" width="40" height="40" :class="$style.icon" />

      <div :class="$style.title" data-test="app-feature-title">{{ feature.title }}</div>
      <div data-test="app-feature-text">{{ feature.text }}</div>

      <FeatureBackground :class="$style.background" />
    </div>
  </div>
</template>

<script setup lang="ts">
import { computed } from 'vue';

import FeatureBackground from '@/common/images/feature-background.svg';

import IconShare from '@/common/icons/share.svg?component';
import IconProgress from '@/common/icons/to-failure.svg?component';
import IconPlan from '@/common/icons/activity.svg?component';

import { useTI18n } from '@/common/composables';

const { t } = useTI18n();

const APP_FEATURES = computed(() => [
  { icon: IconProgress, title: t('features.progressTitle'), text: t('features.progressText') },
  { icon: IconShare, title: t('features.shareTitle'), text: t('features.shareText') },
  { icon: IconPlan, title: t('features.planTitle'), text: t('features.planText') },
]);
</script>

<style module lang="scss">
.features {
  display: flex;
  gap: 32px;
}

.feature {
  position: relative;
  display: flex;
  flex-direction: column;
  gap: 8px;
  align-items: flex-start;
  padding: 32px;
  overflow: hidden;
  border: 1px solid var(--color-gray);
  border-radius: 12px;
}

.icon {
  color: var(--color-primary);
}

.title {
  font-size: 1.25rem;
  font-weight: 700;
  color: var(--color-primary);
  border-bottom: 4px solid;
}

.background {
  position: absolute;
  top: 0;
  right: 0;
  min-width: 100%;
  min-height: 100%;
  opacity: 0.1;
}

@media (max-width: 960px) {
  .features {
    flex-direction: column;
  }

  .feature {
    padding: 24px;
  }
}
</style>
