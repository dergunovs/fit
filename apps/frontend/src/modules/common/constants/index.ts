import IconExercise from '@/common/icons/nav-exercise.svg?component';
import IconActivity from '@/common/icons/nav-activity.svg?component';
import IconEquipment from '@/common/icons/weight.svg?component';
import IconUser from '@/common/icons/nav-user.svg?component';

import { URL_USER } from '@/user/constants';
import { URL_EXERCISE } from '@/exercise/constants';
import { URL_ACTIVITY_ADMIN } from '@/activity/constants';
import { URL_EQUIPMENT } from '@/equipment/constants';
import { INavItem } from '@/common/interface';
import { URL_SETUP } from '@/auth/constants';

export const NAV_ITEMS: INavItem[] = [
  { _id: '1', url: URL_EXERCISE, title: 'Упражнения', icon: IconExercise },
  { _id: '2', url: URL_ACTIVITY_ADMIN, title: 'Занятия', icon: IconActivity },
  { _id: '3', url: URL_EQUIPMENT, title: 'Оборудование', icon: IconEquipment },
  { _id: '4', url: URL_USER, title: 'Пользователи', icon: IconUser },
];

export const UPDATE_BUTTON_TEXT = 'Обновить';
export const CREATE_BUTTON_TEXT = 'Создать';

export const URL_HOME = '/';
export const URL_ERROR = '/404';

export const GUEST_PAGES = [URL_HOME, URL_SETUP];
