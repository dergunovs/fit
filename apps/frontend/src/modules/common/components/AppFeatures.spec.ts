import { describe, it, expect, beforeEach, afterEach } from 'vitest';
import { VueWrapper, enableAutoUnmount } from '@vue/test-utils';
import { dataTest } from 'mhz-helpers';

import AppFeatures from './AppFeatures.vue';

import { wrapperFactory } from '@/common/test';

const feature = dataTest('app-feature');
const title = dataTest('app-feature-title');
const text = dataTest('app-feature-text');

let wrapper: VueWrapper<InstanceType<typeof AppFeatures>>;

beforeEach(() => {
  wrapper = wrapperFactory(AppFeatures);
});

enableAutoUnmount(afterEach);

describe('AppFeatures', async () => {
  it('exists', async () => {
    expect(wrapper.findComponent(AppFeatures)).toBeTruthy();
  });

  it('matches snapshot', async () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('shows features', async () => {
    expect(wrapper.findAll(feature).length).toBe(3);
    expect(wrapper.find(title).text()).toBe('Отслеживайте прогресс');
    expect(wrapper.find(text).text()).toBe(
      'Таблица со статистикой по упражнениям. Графики с динамикой по занятиям, подходам, повторам, времени и группам мышц.'
    );
  });
});
