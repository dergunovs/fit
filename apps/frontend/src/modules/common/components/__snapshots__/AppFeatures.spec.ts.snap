// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AppFeatures > matches snapshot 1`] = `
"<div class="_features_5ff3fc">
  <div class="_feature_5ff3fc" data-test="app-feature">
    <anonymous-stub width="40" height="40" class="_icon_5ff3fc"></anonymous-stub>
    <div class="_title_5ff3fc" data-test="app-feature-title">Отслеживайте прогресс</div>
    <div data-test="app-feature-text">Таблица со статистикой по упражнениям. Графики с динамикой по занятиям, подходам, повторам, времени и группам мышц.</div>
    <anonymous-stub class="_background_5ff3fc"></anonymous-stub>
  </div>
  <div class="_feature_5ff3fc" data-test="app-feature">
    <anonymous-stub width="40" height="40" class="_icon_5ff3fc"></anonymous-stub>
    <div class="_title_5ff3fc" data-test="app-feature-title">Делитесь результатами</div>
    <div data-test="app-feature-text">В один клик скопируйте данные по выполненной тренировке в календаре, включая время каждого подхода.</div>
    <anonymous-stub class="_background_5ff3fc"></anonymous-stub>
  </div>
  <div class="_feature_5ff3fc" data-test="app-feature">
    <anonymous-stub width="40" height="40" class="_icon_5ff3fc"></anonymous-stub>
    <div class="_title_5ff3fc" data-test="app-feature-title">Планируйте тренировки</div>
    <div data-test="app-feature-text">Повторите одну из прошлых тренировок. Предварительный расчет длительности тренировки на основе вашей статистики.</div>
    <anonymous-stub class="_background_5ff3fc"></anonymous-stub>
  </div>
</div>"
`;
